### PR TITLE
Dịch "softmax-regression-scratch_vn" - Phần 4

### LINKED ISSUE #770: Dịch "softmax-regression-scratch_vn" - Phần 4
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/softmax-regression-scratch_vn.md#L217-L327

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by sudohainguyen**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #770 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 1]
```diff
@@ -220,7 +220,7 @@ def cross_entropy(y_hat, y):
 ## Classification Accuracy
 -->
 
-## *dịch tiêu đề phía trên*
+Độ chính xác cho bài toán phân loại
```
Comment (tanduong) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]: Độ chính xác nghe giống precision hơn là accuracy (https://blog.minitab.com/blog/real-world-quality-improvement/accuracy-vs-precision-whats-the-difference). Tính đúng đắn thì lại dễ nhầm với truthfulness.  Thuật ngữ này nên để thảo luận.

Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
## Độ chính xác cho bài toán phân loại
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 2]
```diff
@@ -229,7 +229,10 @@ Gmail must categorize an email into Primary, Social, Updates, or Forums.
 It might estimate probabilities internally, but at the end of the day it has to choose one among the categories.
 -->
 
-*dịch đoạn phía trên*
+Với phân phối xác suất dự đoán `y_hat`, ta thường chọn lớp có xác suất dự đoán cao nhất khi ta cần đầu ra là một dự đoán cụ thể.
+Tuy nhiên, nhiều ứng dụng thực tế cần ta đưa ra sự lựa chọn cụ thể.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Với phân phối xác suất dự đoán `y_hat`, ta thường chọn lớp có xác suất dự đoán cao nhất khi cần đưa ra một dự đoán cụ thể vì nhiều ứng dụng trong thực tế có yêu cầu như vậy.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 3]
```diff
@@ -238,7 +241,10 @@ Although it can be difficult optimize accuracy directly (it is not differentiabl
 it is often the performance metric that we care most about, and we will nearly always report it when training classifiers.
 -->
 
-*dịch đoạn phía trên*
+Các dự đoán được coi là chính xác khi chúng khớp với lớp thực tế `y`.
+Độ chính xác phân loại được tính bởi tỉ lệ các dự đoán chính xác trên tất cả các dự đoán đã đưa ra.
+Dù ta có thể gặp khó khăn khi tối ưu hóa trực tiếp độ chính xác (chúng không không khác biệt),
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Dù ta có thể gặp khó khăn khi tối ưu hóa trực tiếp độ chính xác (chúng không khả vi),
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 4]
```diff
@@ -250,7 +256,13 @@ The result is an `ndarray` containing entries of 0 (false) and 1 (true).
 Taking the mean yields the desired result.
 -->
 
-*dịch đoạn phía trên*
+Độ chính xác được tính toán như sau:
+Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm lấy ra các lớp được dự đoán (cho bởi chỉ số tại hạng mục lớn nhất của mỗi hàng).
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 5]
```diff
@@ -250,7 +256,13 @@ The result is an `ndarray` containing entries of 0 (false) and 1 (true).
 Taking the mean yields the desired result.
 -->
 
-*dịch đoạn phía trên*
+Độ chính xác được tính toán như sau:
+Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm 
+Kết quả trả về sẽ có cùng kích thước với biến `y`
+và bây giờ ta chỉ cần so sánh hai ma trận.
+Vì operator `==` so khớp cả kiểu dữ liệu của biến (ví dụ một biến `int` và một biến `float32` không thể bằng nhau), ta cần phải ép kiểu chúng một cách thống nhất (ở đây ta chọn kiểu `float32`).
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Vì toán tử `==` so khớp cả kiểu dữ liệu của biến (ví dụ một biến `int` và một biến `float32` không thể bằng nhau), ta cần phải ép kiểu chúng một cách thống nhất (ở đây ta chọn kiểu `float32`).
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 6]
```diff
@@ -268,7 +280,10 @@ The second example's prediction category is $2$ (the largest element of the row
 Therefore, the classification accuracy rate for these two examples is $0.5$.
 -->
 
-*dịch đoạn phía trên*
+Ta sẽ tiếp tục sử dụng biến `y_hat` và `y` đã được định nghĩa trong hàm `pick` lần lượt tương ứng với dự đoán từ phân phối xác suất và nhãn.
+Có thể thấy rằng kết quả dự đoán của ví dụ đầu tiên là $2$ (phần từ lớn nhất trong hàng là 0.6 với chỉ số tương ứng là $2$) không khớp với nhãn thực tế là $0$. 
+Dự đoán ở ví dụ thứ hai là $2$ (phần tự lớn nhất hàng là $0.5$ với chỉ số tương ứng là $2$) khớp với nhãn thực tế là $2$.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Dự đoán ở ví dụ thứ hai là $2$ (phần tử lớn nhất hàng là $0.5$ với chỉ số tương ứng là $2$) khớp với nhãn thực tế là $2$.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 7]
```diff
@@ -268,7 +280,10 @@ The second example's prediction category is $2$ (the largest element of the row
...
+Do đó độ chính xác phân loại cho hai ví dụ này là $0.5$
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Do đó độ chính xác phân loại cho hai ví dụ này là $0.5$.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 8]
```diff
@@ -318,7 +333,7 @@ class Accumulator(object):
 Because we initialized the `net` model with random weights, the accuracy of this model should be close to random guessing, i.e., $0.1$ for $10$ classes.
 -->
 
-*dịch đoạn phía trên*
+Vì ta đã khởi tạo trước đó cho mô hình `net` với trọng số ngẫu nhiên nên độ chính xác của mô hình lúc này là một con số ngẫu nhiên, ví dụ: $0.1$ cho $10$ lớp.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Vì ta đã khởi tạo trước đó cho mô hình `net` với trọng số ngẫu nhiên nên độ chính xác của mô hình lúc này sẽ ngang với đoán mò, ví dụ: độ chính xác bằng $0.1$ cho $10$ lớp.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 1 above)
Comment (sudohainguyen) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: dạ em nghĩ dễ xảy ra chiều ngược lại hơn, tức nếu dịch precision là chính xác sẽ hiểu lầm là accuracy ạ.

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 9]
```diff
@@ -318,7 +332,7 @@ class Accumulator(object):
...
```
Comment (sudohainguyen) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Vì ta đã khởi tạo trước đó cho mô hình `net` với trọng số ngẫu nhiên nên độ chính xác của mô hình lúc này như là một dự đoán ngẫu nhiên, ví dụ: độ chính xác bằng $0.1$ cho $10$ lớp.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 10]
```diff
@@ -268,7 +279,10 @@ The second example's prediction category is $2$ (the largest element of the row
 Therefore, the classification accuracy rate for these two examples is $0.5$.
 -->
 
-*dịch đoạn phía trên*
+Ta sẽ tiếp tục sử dụng biến `y_hat` và `y` đã được định nghĩa trong hàm `pick` lần lượt tương ứng với dự đoán từ phân phối xác suất và nhãn.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Ta sẽ tiếp tục sử dụng biến `y_hat` và `y` đã được định nghĩa trong hàm `pick`, lần lượt tương ứng với phân phối xác suất được dự đoán và nhãn.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 11]
```diff
@@ -279,7 +293,7 @@ accuracy(y_hat, y) / len(y)
 Similarly, we can evaluate the accuracy for model `net` on the dataset (accessed via `data_iter`).
 -->
 
-*dịch đoạn phía trên*
+Tương tự, ta có thể đánh giá độ chính xác của mô hình `net` trên bộ dữ liệu (được truy xuất thông qua `data_iter`).
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Tương tự, ta có thể đánh giá độ chính xác của mô hình `net` trên tập dữ liệu (được truy xuất thông qua `data_iter`).
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 12]
```diff
@@ -294,7 +308,7 @@ def evaluate_accuracy(net, data_iter):
 Here `Accumulator` is a utility class to accumulated sum over multiple numbers.
 -->
 
-*dịch đoạn phía trên*
+`Accumulator` ở đây là một lớp đa tiện ích, có tác dụng tính tổng nhiều số.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
`Accumulator` ở đây là một lớp đa tiện ích, có tác dụng tính tổng tích lũy của nhiều số.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 8 above)
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: `i.e.` mình hay dịch là `tức là`, `nghĩa là`.
@sudohainguyen mình thấy suggestion của @lkhphuc sát bản gốc hơn
```suggestion
Vì ta đã khởi tạo mô hình `net` với trọng số ngẫu nhiên nên độ chính xác của mô hình lúc này sẽ ngang với việc đoán mò, tức độ chính xác bằng $0.1$ với $10$ lớp.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 13]
```diff
@@ -250,7 +255,13 @@ The result is an `ndarray` containing entries of 0 (false) and 1 (true).
 Taking the mean yields the desired result.
 -->
 
-*dịch đoạn phía trên*
+Độ chính xác được tính toán như sau:
+Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm lấy ra các lớp được dự đoán (cho bởi chỉ số tại hạng mục lớn nhất của mỗi hàng).
+Kết quả trả về sẽ có cùng kích thước với biến `y`
+và bây giờ ta chỉ cần so sánh hai ma trận.
+Vì toán tử `==` so khớp cả kiểu dữ liệu của biến (ví dụ một biến `int` và một biến `float32` không thể bằng nhau), ta cần phải ép kiểu chúng một cách thống nhất (ở đây ta chọn kiểu `float32`).
+Kết quả sẽ là một biến `ndarray` chứa các giá trị 0 (false) và 1 (true).
```
[the remark this answers is not in the document]
Comment (sudohainguyen) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: hay là "Kết quả sẽ có kiểu `ndarray` bao gồm các giá trị 0 (false) và 1 (true)."

[the remark this answers is not in the document]
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: `ndarray` là 1 object mà, mình thấy dịch vậy cũng được. Với cả trong chapter ndarray cũng hay dịch như vậy rồi.

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 1 above)
Comment (tanduong) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Như anh @tiepvupsu có comment là nào giờ nhiều sách dịch `accuracy` là `độ chính xác` rồi nên để vậy hợp lý hơn.

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: sao không dịch true false luôn nhỉ @minhduc0711 ?

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 9 above)
Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: chỗ này không phải số ngẫu nhiên đâu bạn. `0.1` là độ chính xác của việc đoán ngẫu nhiên. Nếu có 100 classes thì độ chính xác ban đầu sẽ là `0.01`.

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: @lkhphuc chắc tác giả muốn imply khi gọi hàm sum() thì array boolean sẽ được cast về int, mình nghĩ nên để nguyên như thế

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 14]
```diff
@@ -220,7 +220,7 @@ def cross_entropy(y_hat, y):
 ## Classification Accuracy
 -->
 
-## *dịch tiêu đề phía trên*
+## Độ chính xác cho bài toán phân loại
```
Comment (duythanhvn) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
## Độ chính xác cho bài toán Phân loại
```
